code/example8.py

```diff
@@ -47,7 +47,17 @@ def main(stdscr):
 
 		# set direction if user pressed any arrow key
 		if key in [curses.KEY_RIGHT, curses.KEY_LEFT, curses.KEY_DOWN, curses.KEY_UP]:
-			direction = key
+			# store the previous snake's direction, in order to
+			# avoid that the new direction is the opposite of
+			# the previous one on the same axis.
+			# E.g.: the new direction can't be up/down/left/right
+			# if the previous direction was down/up/right/left
+			prev_direction = direction
+			if (key == curses.KEY_UP and prev_direction != curses.KEY_DOWN or
+		    		key == curses.KEY_DOWN and prev_direction != curses.KEY_UP or
+	            		key == curses.KEY_RIGHT and prev_direction != curses.KEY_LEFT or
+	            		key == curses.KEY_LEFT and prev_direction != curses.KEY_RIGHT):
+				direction = key
```

Comment:
Let us keep a dictionary of opposite directions so that we do not have to do so many comparisons everytime.
```py
OPPOSITE_DIRECTION_DICT = {
    curses.KEY_UP: curses.KEY_DOWN,
    curses.KEY_DOWN: curses.KEY_UP,
    curses.KEY_RIGHT: curses.KEY_LEFT,
    curses.KEY_LEFT: curses.KEY_RIGHT
}

DIRECTIONS_LIST = [curses.KEY_RIGHT, curses.KEY_LEFT, curses.KEY_DOWN, curses.KEY_UP]
```

```py
if key in DIRECTIONS_LIST and key != OPPOSITE_DIRECTION_DICT[direction]:
     direction = key
```